sbi/inference/snpe/snpe_base.py

```diff
@@ -269,6 +227,18 @@ def _run_simulations(
 
         return theta, x, self._mask_sims_from_prior(round_, theta.size(0))
 
+    def _z_score_embedding(self, x: Tensor) -> nn.Module:
+        """Return embedding net with a standardizing step preprended."""
+
+        # XXX Mouthful, rename self.posterior.nn
+        embed_nn = self._neural_posterior.neural_net._embedding_net
+
+        nonzero_std = torch.clamp(torch.std(x, dim=0), min=self._z_score_min_std)
+        preprocess = Standardize(torch.mean(x, dim=0), nonzero_std)
```

Comment:
Hm. I think I would rather use `self._z_score_min_std` only if `torch.std(x, dim=0)==0`. It's possible that a feature takes only very very small numbers, possibly much smaller than `1e-7` (which is default for `_z_score_min_std`) . In this case this would not behave as wanted.

Reply:
That's a really good point!